app/uk/gov/hmrc/agentfiaccesscontrol/controllers/PayeController.scala

```
@@ -44,7 +44,8 @@ class PayeController @Inject()(relationshipConnector: RelationshipConnector,
           case 200 =>
             auditService.sendAccessRequestAuditEvent(setAuditData(authProviderId, arnValue, agentCode, clientId, accessGrantedTrue))
             Ok
-          case _ ⇒
+        } recover {
+          case _ =>
```

@Smudgey It's not considered good practice to catch all exceptions, it's better to catch the specific exception you have thought about. Would you really want to return NotFound if a NullPointerException or OutOfMemory error happened, or would it be better to leave the exception unhandle (which would result in a 500 response and the exception stack trace being logged).

Please can you change this to catch a specific exception.